File: 2717-semi-ordered-permutation/2717-semi-ordered-permutation.py
class Solution {
public:
    int semiOrderedPermutation(vector<int>& A) {
        int n = A.size();
        int i = find(A.begin(), A.end(), 1) - A.begin();
        int j = find(A.begin(), A.end(), n) - A.begin();
        return i + n - 1 - j - (i > j);
    }
};

File: 2717-semi-ordered-permutation/2717-semi-ordered-permutation.js
var semiOrderedPermutation = function(nums) {
    const n = nums.length;
    let fi = -1, li = -1;
    for (let i = 0; i < n; i++) {
        if (nums[i] === 1) {
            fi = i;
        } else if (nums[i] === n) {
            li = i;
        }
    }
    let ans = fi - 0 + n - 1 - li;
    if (li < fi) {
        ans--;
    }
    return ans;
};